src/Microsoft.ML.Data/Evaluators/AnomalyDetectionEvaluator.cs

```diff
@@ -97,15 +97,15 @@ private protected override void CheckScoreAndLabelTypes(RoleMappedSchema schema)
             var t = score.Type;
             if (t != NumberType.Float)
                 throw Host.Except("Score column '{0}' has type '{1}' but must be R4", score, t).MarkSensitive(MessageSensitivity.Schema);
-            Host.Check(schema.Label != null, "Could not find the label column");
-            t = schema.Label.Type;
+            Host.Check(schema.Label.HasValue, "Could not find the label column");
+            t = schema.Label.Value.Type;
             if (t != NumberType.Float && t.KeyCount != 2)
-                throw Host.Except("Label column '{0}' has type '{1}' but must be R4 or a 2-value key", schema.Label.Name, t).MarkSensitive(MessageSensitivity.Schema);
+                throw Host.Except("Label column '{0}' has type '{1}' but must be R4 or a 2-value key", schema.Label.Value.Name, t).MarkSensitive(MessageSensitivity.Schema);
```

Comment:
>R4  [](start = 81, length = 3)

sidenote, does it make sense to start using 'float' etc. in the user errors, instead of the internal types?

Reply:
nvm, I'll log an issue and we can discuss there, rather than comments on  this PR. 

---
In reply to: [243381184](https://github.com/dotnet/machinelearning/pull/1924#discussion_r243381184) [](ancestors = 243381184)

Reply:
It may. In fact I think we've discussed this as a desirable outcome, but if we do it we should do it *everywhere* as part of a deliberate policy. So your plan to open an issue seems good to me, @sfilipi.